lib/kracken/controllers/token_authenticatable.rb

```diff
@@ -0,0 +1,61 @@
+module Kracken
+  module Controllers
+    module TokenAuthenticatable
+      def self.included(base)
+        base.define_singleton_method(:realm) do |realm = nil|
+          realm ||= (superclass.try(:realm) || 'Application')
+          @realm = realm
+        end
+
+        base.instance_exec do
+          before_action :authenticate_user_with_token!
+          helper_method :current_user
+        end
+      end
+
+      private
+
+      # `authenticate_or_request_with_http_token` is a nice Rails helper:
+      # http://api.rubyonrails.org/classes/ActionController/HttpAuthentication/Token/ControllerMethods.html#method-i-authenticate_or_request_with_http_token
+      def authenticate_user_with_token!
+        munge_header_auth_token!
+
+        authenticate_or_request_with_http_token(realm) { |token, _options|
+          # Attempt to reduce namespace conflicts with controllers which may access
+          # an team instance for display.
+          @_current_user = Authenticator.user_with_token(token)
+        }
+      end
+
+      def current_user
+        @_current_user
+      end
```

Comment:
Any reason to use `@_current_user` over `@current_user`? I think I original did that in the other repo to not accidentally conflict with this one. But maybe now it's not needed and better to be consistent through the gem.